### PR TITLE
added text classification in nlp_apps

### PR DESCRIPTION
**Pull request by sagar-sehgal**
In this PR, I have added Text Classification using Naive Bayes Classifier in `nlp_apps.ipynb` as was discussed in https://github.com/aimacode/aima-python/issues/890 . It contains

- Text Classification explanation
- Applications under Text Classification
- Word Sense Disambiguation as an application of Text Classification
- 5 steps to explaining the procedure of Text Classification

### CODE REVIEW

File: nlp_apps.ipynb [hunk 1]
```diff
@@ -797,11 +772,251 @@
...
+    "As you might have observed that our dataset is equally balanced i.e. we have an equal number of words in both the classes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Step 4:-Making the Naive Bayes Model**\n",
+    "\n",
+    "Using Naive Bayes classifier we can calculate the probability of a word in `company` and `fruit` class and then multiplying all of them to get the probability of that sentence belonging each of the given classes. But if a word is not in our dictionary then this leads to the probability of that class becoming zero. For eg:- the word *Foxconn* is not in the dictionary of any of the classes. Due to this the \n",
```
Comment (MrDupin) [inline nlp_apps.ipynb]: Add a 'the' before 'Naive Bayes'. Also, add a 'to' before 'each of the given classes'. Also, you don't need to write 'eg:-', just write 'example'.

Finally, the paragraph seems to end abruptly.

File: nlp_apps.ipynb [hunk 2]
```diff
@@ -797,11 +772,251 @@
...
+    "\n",
+    "To solve the problem we need to use **smoothing**, i.e. providing a minimum non-zero threshold probability to every word that we come across.\n",
+    "\n",
+    "The `UnigramWordModel` class has implemented smoothing by taking an additional argument from the user i.e. the minimum frequency that we would be giving to every word even if it is new to the dictionary."
```
Comment (MrDupin) [inline nlp_apps.ipynb]: Add a comma before 'i.e.'.

File: nlp_apps.ipynb [hunk 3]
```diff
@@ -797,11 +772,251 @@
...
+   "source": [
+    "model_words_0 = UnigramWordModel(words_0, 1)\n",
+    "model_words_1 = UnigramWordModel(words_1, 1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we would be making the Naive Bayes model. For that, we would be making `dist` as we had done earlier in the Authorship Recognition Task."
```
Comment (MrDupin) [inline nlp_apps.ipynb]: I prefer the use of a word other than 'making'. It is a bit too general.

File: nlp_apps.ipynb [hunk 4]
```diff
@@ -797,11 +772,251 @@
...
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from learning import NaiveBayesLearner\n",
+    "\n",
+    "dist = {('company', 1): model_words_0, ('fruit', 1): model_words_1}\n",
+    "\n",
+    "nBS = NaiveBayesLearner(dist, simple = True)"
```
Comment (MrDupin) [inline nlp_apps.ipynb]: No spaces are needed around the '=' when passing arguments.

File: nlp_apps.ipynb [hunk 5]
```diff
@@ -797,11 +772,251 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Step 5:- Predict the class of the label**\n",
+    "\n",
+    "Now we would be making a function that does pre-procesing of the sentences which we have taken for testing. And then predicting the class of every sentence in the document."
```
Comment (MrDupin) [inline nlp_apps.ipynb]: It is 'will' instead of 'would'. Also, 'procesing' has two 's'. Again, I would prefer the use of another verb other than 'making'; in this case 'writing' works fine.

File: nlp_apps.ipynb [hunk 6]
```diff
@@ -797,11 +772,251 @@
...
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Apple Inc. supplier Foxconn demos its own iPhone-compatible smartwatch\t-company\n",
+      "I now know how to make a delicious apple pie thanks to the best teachers ever\t-fruit\n"
+     ]
+    }
+   ],
+   "source": [
+    "# prediction the class of every sentence in the test set\n",
```
Comment (MrDupin) [inline nlp_apps.ipynb]: What do you mean by 'prediction the class of every sentence in the test set'? The sentence is no grammatically correct and it is a bit confusing.

File: nlp_apps.ipynb [hunk 7]
```diff
@@ -797,11 +772,251 @@
...
+    "for i in test_data:\n",
+    "    print(i + \"\\t-\" + recognize(i, nBS))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You might have observed that our predictions are correct. Though they might not give correct results because of lack of data. And we are clearly able to differentiate between sentences in a much better way. \n",
```
Comment (MrDupin) [inline nlp_apps.ipynb]: This is a bit awkward to read. Can you rephrase/rewrite it?

File: nlp_apps.ipynb [hunk 8]
```diff
@@ -797,11 +772,251 @@
...
+    "\n",
+    "As you might have observed that the above method is very much similar to the Authorship prediction, which is also a type of Text Classification. Like this most of Text Classification have the same underlying structure and follow a similar procedure."
```
Comment (MrDupin) [inline nlp_apps.ipynb]: You don't need the 'that' after 'observed'; a comma works better.

Comment (sagar-sehgal) [reply nlp_apps.ipynb]: Thanks, @MrDupin for such a close review of the PR. Working on the changes requested by you. Thanks!
